apiserver/facades/client/storage/shim.go

```diff
@@ -66,11 +69,14 @@ func NewFacadeV3(
 	registry := stateenvirons.NewStorageProviderRegistry(env)
 	pm := poolmanager.New(state.NewStateSettings(st), registry)
 
-	storageAccessor, err := getStorageAccessor(st)
+	storageAccessor, storageVolume, storageFile, err := getStorageAccessors(st)
 	if err != nil {
 		return nil, errors.Annotate(err, "getting backend")
 	}
-	return NewAPIv3(stateShim{st}, storageAccessor, registry, pm, resources, authorizer)
+	return NewAPIv3(
+		stateShim{st},
+		storageAccessor, storageVolume, storageFile,
+		registry, pm, resources, authorizer)
 }
 
 type storageAccess interface {
```

Comment:
Actually, this naming isn't too bad either (if you don't like Instances, Filesystems, Volumes): `InstanceAccess, FilesystemAccess, VolumeAccess`?

Reply:
+1 I like the 'access' suffix here...